API/src/main/java/backend/API/FileUploadController.java

```diff
@@ -226,11 +226,12 @@ public ResponseEntity<ArrayList<fileTimespan>> listFolderTimespans(
         Object[] container = {null, null};
         paths.forEach(
             path -> {
-              if (path.getParent() != null) {
+              Path parent = path.getParent();
+              if (storageService.canComputeTimespan(parent.toString())) {
```

Comment:
If parent is null, error is thrown. Needs to check this before invoking .toString() method. This would activate if you upload a CSV file alone.

Reply:
Resolved with 49f9ef799930ef3c8d302104fb1e5249a2915043